core/src/network/mod.rs

```diff
@@ -298,8 +298,10 @@ pub mod tests {
         assert_eq!(links.len(), 2, "links = {:?}", links);
         // can be in any order
         assert!(
-            (links[0] == entry_addresses[1] || links[0] == entry_addresses[2])
-                && (links[1] == entry_addresses[1] || links[1] == entry_addresses[2])
+            (links[0] == (entry_addresses[1].clone(), CrudStatus::Live)
+                || links[0] == (entry_addresses[2].clone(), CrudStatus::Live))
+                && (links[1] == (entry_addresses[1].clone(), CrudStatus::Live)
+                    || links[1] == (entry_addresses[2].clone(), CrudStatus::Live))
```

Comment:
This test is getting heavy. If it fails it would be probably be a pain to debug. I think it's outside the scope of your PR to fix it however.

Reply:
Yes it would be nice to break out in follow up pr